src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/tools/HeapSummary.java

```diff
@@ -86,10 +86,8 @@ public void run() {
       printValMB("MetaspaceSize            = ", getFlagValue("MetaspaceSize", flagMap));
       printValMB("CompressedClassSpaceSize = ", getFlagValue("CompressedClassSpaceSize", flagMap));
       printValMB("MaxMetaspaceSize         = ", getFlagValue("MaxMetaspaceSize", flagMap));
-      if (heap instanceof ShenandoahHeap) {
-         printValMB("ShenandoahRegionSize     = ", ShenandoahHeapRegion.regionSizeBytes());
-      } else {
-         printValMB("G1HeapRegionSize         = ", HeapRegion.grainBytes());
+      if (heap instanceof G1CollectedHeap) {
+        printValMB("G1HeapRegionSize         = ", HeapRegion.grainBytes());
```

Comment:
Nit: I'd suggest to move this line to the `G1CollectedHeap` specific block after line 123. It would be consistent with your change for `ShenandoahRegionSize`.

Reply:
Thought about it, but decided that this thing is about the flags, so kept it as is.